>From 3b3ff8bfb04224866b7573e0bf620299313ca9b4 Mon Sep 17 00:00:00 2001
From: William Chargin <wchargin@gmail.com>
Date: Tue, 31 Mar 2020 19:39:13 -0700
Subject: [PATCH] monster: permit spawning on same row as player

Due to a typo, the `(row == rogue.row) && (col = rogue.col)` check is
effectively equivalent to `(row == rogue.row)`. This prevents monsters
from spawning on the same row as the player when a scroll of create
monster is read.

To reproduce, play Rogue until you find a scroll of create monster, then
enter a horizontal tunnel such that the only adjacent spaces are in the
same row as the player. Read the scroll, and note that you "hear a faint
cry of anger in the distance".
---
 rogue/monster.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rogue/monster.c b/rogue/monster.c
index 2bcc25d..30b0705 100644
--- a/rogue/monster.c
+++ b/rogue/monster.c
@@ -688,7 +688,7 @@ create_monster()
 
 	for (i = 0; i < 9; i++) {
 		rand_around(i, &row, &col);
-		if (((row == rogue.row) && (col = rogue.col)) ||
+		if (((row == rogue.row) && (col == rogue.col)) ||
 				(row < MIN_ROW) || (row > (DROWS-2)) ||
 				(col < 0) || (col > (DCOLS-1))) {
 			continue;
-- 
2.26.0

